tests/Runtime-Microsoft.Android.Sdk/Microsoft.Android-Tests.csproj

```diff
@@ -0,0 +1,148 @@
+
+  <ItemGroup>
+    <ProguardConfiguration Include="$(_MonoAndroidTest)proguard.cfg" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <AndroidResource Include="$(_MonoAndroidTest)Resources\drawable\Icon.png">
+      <Link>Resources\drawable\Icon.png</Link>
+    </AndroidResource>
+    <AndroidResource Include="$(_MonoAndroidTest)Resources\drawable\AndroidPressed.png">
+      <Link>Resources\drawable\AndroidPressed.png</Link>
+    </AndroidResource>
+    <AndroidResource Include="$(_MonoAndroidTest)Resources\drawable\android_focused.png">
+      <Link>Resources\drawable\android_focused.png</Link>
+    </AndroidResource>
+    <AndroidResource Include="$(_MonoAndroidTest)Resources\drawable\android_normal.png">
+      <Link>Resources\drawable\android_normal.png</Link>
+    </AndroidResource>
+    <AndroidResource Include="$(_MonoAndroidTest)Resources\drawable\android_button.xml">
+      <Link>Resources\drawable\android_button.xml</Link>
+    </AndroidResource>
+    <AndroidResource Include="$(_MonoAndroidTest)Resources\layout\uppercase_custom.axml">
+      <Link>Resources\layout\uppercase_custom.axml</Link>
+    </AndroidResource>
+    <AndroidResource Include="$(_MonoAndroidTest)Resources\layout\lowercase_custom.axml">
+      <Link>Resources\layout\lowercase_custom.axml</Link>
+    </AndroidResource>
+    <AndroidResource Include="$(_MonoAndroidTest)Resources\xml\XmlReaderResourceParser.xml">
+      <Link>Resources\xml\XmlReaderResourceParser.xml</Link>
+    </AndroidResource>
+    <AndroidResource Include="$(_MonoAndroidTest)Resources\layout\FragmentFixup.axml">
+      <Link>Resources\layout\FragmentFixup.axml</Link>
+    </AndroidResource>
+    <AndroidBoundLayout Include="$(_MonoAndroidTest)Resources\layout\Main.axml">
+      <Link>Resources\layout\Main.axml</Link>
+    </AndroidBoundLayout>
+  </ItemGroup>
+
+  <ItemGroup>
+    <AndroidAsset Include="$(_MonoAndroidTest)Assets\asset1.txt">
+      <Link>Assets\asset1.txt</Link>
+    </AndroidAsset>
+    <AndroidAsset Include="$(_MonoAndroidTest)Assets\subfolder\asset2.txt">
+      <Link>Assets\subfolder\asset2.txt</Link>
+    </AndroidAsset>
+    <AndroidAsset Include="$(_MonoAndroidTest)Assets\subfolder\accept_request.png">
+      <Link>Assets\subfolder\accept_request.png</Link>
+    </AndroidAsset>
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Xamarin.Android.NUnitLite\Xamarin.Android.NUnitLite.NET.csproj" />
+    <ProjectReference Include="..\TestRunner.Core\TestRunner.Core.NET.csproj" />
+    <ProjectReference Include="..\TestRunner.NUnit\TestRunner.NUnit.NET.csproj" />
+    <ProjectReference Include="$(_MonoAndroidTest)Mono.Android-Test.Library\Mono.Android-Test.Library.NET.csproj" />
+    <ProjectReference Include="$(_MonoAndroidTest)Java.Interop-Tests\Java.Interop-Tests.NET.csproj" />
+  </ItemGroup>
+
+  <Import Project="Microsoft.Android-Tests.projitems" />
+  <Import Project="..\..\build-tools\scripts\TestApks.targets" />
+
+  <Target Name="GenerateNuGetConfig" >
```

Comment:
*If* this file is added to `Xamarin.Android-Tests.sln`, could you please test the following?

1. Within Visual Studio for Mac,
2. Load `Xamarin.Android-Tests.sln`.
3. Add a new project to the `.sln`.

What happens?

*Normally*, what would happen. is that *every `.csproj` in the `.sln`* is "re-formatted", screwing up whitespace, and requiring a `git checkout -f HEAD …` to clean things up.

I don't know if this is still the case for .NET 5+ projects/SDK short-form projects.  This is the case for "normal" projects.

It would likely be safer to move these `<Target/>`s into `Directory.Build.targets`.

Reply:
I don't think we should add this to `Xamarin.Android-Tests.sln` at the moment, as it will break our macOS build job. These new project files will only compile against a specific version of .NET 5 that also has our workload installed into it. Additionally, I think there is still IDE work underway that will be needed to be able to load and use a project with a target framework of `net5.0-android`. I will move the targets over, but I don't think we should include any `net5.0-android` projects in our legacy solutions, or if we do want to do so eventually we should at least wait until there is a .NET 6 public preview available that supports it.